src/Shared/FileUtilities.GetFolderPath.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Build.Shared
+{
+    internal static partial class FileUtilities
+    {
+        //  .NET Core doesn't currently have APIs to get the "special" folders (ie the ones defined in the Environment.SpecialFolder enum)
+        //  The code here is mostly copied out of the .NET reference source for this functionality
```

Comment:
@akoeplinger Excellent point!

This PR is provided as part of my job at Microsoft, who is the copyright holder on the code. By providing this PR, we're licensing the code submitted in this PR under the license of this project, which is MIT. 

In this case, the code copied from the reference source was already part of the MIT-licensed reference source at GitHub.